src/datatypes/validation/gadget/PoisonPills.cpp

```diff
@@ -0,0 +1,50 @@
+/**
+ * @file   PoisonPills.cpp
+ * @author Marco Deuscher
+ * @date   28.04.2020 (creation)
+ * @brief  Implemenation of poison pills gadget validation.
+ */
+
+#include "datatypes/gameplay/Movement.hpp"
+#include "GadgetValidator.hpp"
+
+namespace spy::gameplay {
+    bool GadgetValidator::validatePoisonPills(const State &s, GadgetAction a) {
+        using spy::gadget::GadgetEnum;
+
+        // check if target inside map
+        if (!s.getMap().isInside(a.getTarget())) {
+            return false;
+        }
+
+        // check if character has poisonous pills
+        auto character = s.getCharacters().findByUUID(a.getCharacterId().value());
+        auto gadgets = character->getGadgets();
+        auto gadget = std::find_if(gadgets.begin(), gadgets.end(), [](const gadget::Gadget &g) {
+            return g.getType() == GadgetEnum::POISON_PILLS;
+        });
```

Comment:
Vielleicht ist es sinnvoll diesen code in eine helper-methode (`hasCocktail(const State&, UUID, GadgetEnum)` oder so) auszulagern

Reply:
Jo ist sinnvoll.
Wo sollen solche helper Funktionen implementiert werden?
Ist vllt auch auch sinnvolle eine zu machen, die einen Character zurück gibt, gegeben Koordiaten usw.

Reply:
Ja das mit dem character habe ich vorhin auch machen wollen, habe es dann aber sein lassen weil ich optional-referenzen wollte aber das gibts nicht...

Ich würde mal sagen da ist eine neue Util-Klasse angebracht.

Reply:
Ich hab jetzt zwei Funktionen hinzugefügt, eine die auf einen Cocktail prüft und eine die prüft, ob ein Character ein Gadget hat. 
Waren die Funktionen, die ich bisher am häufigsten gebraucht habe.